ICU 59.1 breaks compilation

GCC 7.1 + libboost 1.64 + wxwidgets 3.0.2 :+1: 

```
[51s] g++ -DPACKAGE_NAME=\"poedit\" -DPACKAGE_TARNAME=\"poedit\" -DPACKAGE_VERSION=\"2.0.2\" -DPACKAGE_STRING=\"poedit\ 2.0.2\" -DPACKAGE_BUGREPORT=\"help@poedit.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"poedit\" -DVERSION=\"2.0.2\" -DHAVE_CXX14=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_THREAD=/\*\*/ -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BOOST_IOSTREAMS=/\*\*/ -DHAVE_MKDTEMP=1 -DHAVE_CLD2_PUBLIC_COMPACT_LANG_DET_H=1 -DHAVE_CLD2=1 -I.     -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -fno-strict-aliasing -pthread -I/usr/include -I/usr/lib64/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DwxDEBUG_LEVEL=0 -DWXUSINGDLL -D__WXGTK__ -pthread  -pthread -I/usr/include/gtkspell-2.0 -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/lucene++ -Wall -Wextra  -DwxNO_UNSAFE_WXSTRING_CONV=1 "-DPOEDIT_PREFIX=\"/usr\"" -c -o catalog.o catalog.cpp
[   56s] In file included from /usr/include/wx-3.0/wx/strconv.h:17:0,                                                  
[   56s]                  from /usr/include/wx-3.0/wx/strvararg.h:20,                                                  
[   56s]                  from /usr/include/wx-3.0/wx/string.h:46,                                                     
[   56s]                  from language.h:29,              
[   56s]                  from catalog.h:29,               
[   56s]                  from language.h:29,              
[   56s]                  from catalog.h:29,               
[   56s]                  from cat_sorting.h:29,           
[   56s]                  from cat_sorting.cpp:27:         
[   56s] /usr/include/wx-3.0/wx/buffer.h: In instantiation of 'static const wxScopedCharTypeBuffer<T> wxScopedCharTypeBuffer<T>::CreateOwned(wxScopedCharTypeBuffer<T>::CharType*, size_t) [with T = char16_t; wxScopedCharTypeBuffer<T>::CharType = char16_t; size_t = long unsigned int]':
[   56s] str_helpers.h:176:37:   required from here        
[   56s] /usr/include/wx-3.0/wx/buffer.h:99:27: error: invalid conversion from 'wxScopedCharTypeBuffer<char16_t>::CharType* {aka char16_t*}' to 'const wxChar16* {aka const short unsigned int*}' [-fpermissive]
[   56s]              len = wxStrlen(str);                 
[   56s]                    ~~~~~~~~^~~~~                  
[   56s] In file included from /usr/include/wx-3.0/wx/string.h:45:0,                                                   
[   56s]                  from language.h:29,              
[   56s]                  from catalog.h:29,               
[   56s]                  from cat_sorting.h:29,           
[   56s]                  from cat_sorting.cpp:27:         
[   56s] /usr/include/wx-3.0/wx/wxcrtbase.h:678:32: note:   initializing argument 1 of 'size_t wxStrlen(const wxChar16*)'                                                 
[   56s]         WXDLLIMPEXP_BASE size_t wxStrlen(const wxChar16 *s );                                                 
[   56s]                                 ^~~~~~~~          
[   56s] make[1]: *** [Makefile:696: cat_sorting.o] Error 1                                                            
[   56s] make[1]: *** Waiting for unfinished jobs....      
[   68s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/poedit-2.0.2/src'                                     

```

```
[   21s] g++ -DPACKAGE_NAME=\"poedit\" -DPACKAGE_TARNAME=\"poedit\" -DPACKAGE_VERSION=\"2.0.2\" -DPACKAGE_STRING=\"poedit\ 2.0.2\" -DPACKAGE_BUGREPORT=\"help@poedit.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"poedit\" -DVERSION=\"2.0.2\" -DHAVE_CXX14=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_THREAD=/\*\*/ -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BOOST_IOSTREAMS=/\*\*/ -DHAVE_MKDTEMP=1 -DHAVE_CLD2_PUBLIC_COMPACT_LANG_DET_H=1 -DHAVE_CLD2=1 -I.     -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -fno-strict-aliasing -pthread -I/usr/include -I/usr/lib64/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DwxDEBUG_LEVEL=0 -DWXUSINGDLL -D__WXGTK__ -pthread  -pthread -I/usr/include/gtkspell-2.0 -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/lucene++ -Wall -Wextra  -DwxNO_UNSAFE_WXSTRING_CONV=1 "-DPOEDIT_PREFIX=\"/usr\"" -c -o catalog.o catalog.cpp
[   26s] In file included from /usr/include/wx-3.0/wx/strconv.h:17:0,
[   26s]                  from /usr/include/wx-3.0/wx/strvararg.h:20,
[   26s]                  from /usr/include/wx-3.0/wx/string.h:46,
[   26s]                  from language.h:29,
[   26s]                  from catalog.h:29,
[   26s]                  from cat_sorting.h:29,
[   26s]                  from cat_sorting.cpp:27:
[   26s] /usr/include/wx-3.0/wx/buffer.h: In instantiation of 'static const wxScopedCharTypeBuffer<T> wxScopedCharTypeBuffer<T>::CreateOwned(wxScopedCharTypeBuffer<T>::CharType*, size_t) [with T = char16_t; wxScopedCharTypeBuffer<T>::CharType = char16_t; size_t = long unsigned int]':
[   26s] str_helpers.h:181:37:   required from here
[   26s] /usr/include/wx-3.0/wx/buffer.h:99:27: error: invalid conversion from 'wxScopedCharTypeBuffer<char16_t>::CharType* {aka char16_t*}' to 'const wxChar16* {aka const short unsigned int*}' [-fpermissive]
[   26s]              len = wxStrlen(str);
[   26s]                    ~~~~~~~~^~~~~
[   26s] In file included from /usr/include/wx-3.0/wx/string.h:45:0,
[   26s]                  from language.h:29,
[   26s]                  from catalog.h:29,
[   26s]                  from cat_sorting.h:29,
[   26s]                  from cat_sorting.cpp:27:
[   26s] /usr/include/wx-3.0/wx/wxcrtbase.h:678:32: note:   initializing argument 1 of 'size_t wxStrlen(const wxChar16*)'
[   26s]         WXDLLIMPEXP_BASE size_t wxStrlen(const wxChar16 *s );
[   26s]                                 ^~~~~~~~
[   26s] make[1]: *** [Makefile:696: cat_sorting.o] Error 1
[   26s] make[1]: *** Waiting for unfinished jobs....
[   37s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/poedit-2.0.2/src'
[   37s] make: *** [Makefile:476: all-recursive] Error 1
[   37s] error: Bad exit status from /var/tmp/rpm-tmp.YXt6VL (%build)
[   37s] 
[   37s] 
[   37s] RPM build errors:
[   37s]     Bad exit status from /var/tmp/rpm-tmp.YXt6VL (%build)
[   37s] 
[   37s] argusek.suse.cz failed "build poedit.spec" at Thu Jul  6 22:15:33 UTC 2017.
[   37s] 

```

better , fail later with:

```
[  271s] g++ -DPACKAGE_NAME=\"poedit\" -DPACKAGE_TARNAME=\"poedit\" -DPACKAGE_VERSION=\"2.0.2\" -DPACKAGE_STRING=\"poedit\ 2.0.2\" -DPACKAGE_BUGREPORT=\"help@poedit.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"poedit\" -DVERSION=\"2.0.2\" -DHAVE_CXX14=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_THREAD=/\*\*/ -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BOOST_IOSTREAMS=/\*\*/ -DHAVE_MKDTEMP=1 -DHAVE_CLD2_PUBLIC_COMPACT_LANG_DET_H=1 -DHAVE_CLD2=1 -I.     -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -pthread -I/usr/include -I/usr/lib64/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DwxDEBUG_LEVEL=0 -DWXUSINGDLL -D__WXGTK__ -pthread  -pthread -I/usr/include/gtkspell-2.0 -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/lucene++ -Wall -Wextra  -DwxNO_UNSAFE_WXSTRING_CONV=1 "-DPOEDIT_PREFIX=\"/usr\"" -c -o text_control.o text_control.cpp
[  292s] g++ -DPACKAGE_NAME=\"poedit\" -DPACKAGE_TARNAME=\"poedit\" -DPACKAGE_VERSION=\"2.0.2\" -DPACKAGE_STRING=\"poedit\ 2.0.2\" -DPACKAGE_BUGREPORT=\"help@poedit.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"poedit\" -DVERSION=\"2.0.2\" -DHAVE_CXX14=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_THREAD=/\*\*/ -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BOOST_IOSTREAMS=/\*\*/ -DHAVE_MKDTEMP=1 -DHAVE_CLD2_PUBLIC_COMPACT_LANG_DET_H=1 -DHAVE_CLD2=1 -I.     -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -pthread -I/usr/include -I/usr/lib64/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DwxDEBUG_LEVEL=0 -DWXUSINGDLL -D__WXGTK__ -pthread  -pthread -I/usr/include/gtkspell-2.0 -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/lucene++ -Wall -Wextra  -DwxNO_UNSAFE_WXSTRING_CONV=1 "-DPOEDIT_PREFIX=\"/usr\"" -c -o unicode_helpers.o unicode_helpers.cpp
[  298s] unicode_helpers.cpp: In function 'TextDirection bidi::get_base_direction(const wxString&)':
[  298s] unicode_helpers.cpp:40:19: error: 'to_icu_raw' is not a member of 'str'
[  298s]      auto s = str::to_icu_raw(text);
[  298s]                    ^~~~~~~~~~
[  298s] make[1]: *** [Makefile:687: unicode_helpers.o] Error 1
[  298s] make[1]: *** Waiting for unfinished jobs....
[  304s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/poedit-2.0.2/src'
[  304s] make: *** [Makefile:466: all-recursive] Error 1
[  304s] error: Bad exit status from /var/tmp/rpm-tmp.EMV62x (%build)
[  304s] 
[  304s] 
[  304s] RPM build errors:
[  304s]     Bad exit status from /var/tmp/rpm-tmp.EMV62x (%build)
[  304s] 
[  304s] argusek failed "build poedit.spec" at Fri Jul  7 17:17:44 UTC 2017.
[  304s] 

The buildroot was: /var/tmp/build-root/openSUSE_Factory-x86_64

```

@vslavik 

worse, 
```

[   25s] g++ -DPACKAGE_NAME=\"poedit\" -DPACKAGE_TARNAME=\"poedit\" -DPACKAGE_VERSION=\"2.0.2\" -DPACKAGE_STRING=\"poedit\ 2.0.2\" -DPACKAGE_BUGREPORT=\"help@poedit.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"poedit\" -DVERSION=\"2.0.2\" -DHAVE_CXX14=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_THREAD=/\*\*/ -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BOOST_IOSTREAMS=/\*\*/ -DHAVE_MKDTEMP=1 -DHAVE_CLD2_PUBLIC_COMPACT_LANG_DET_H=1 -DHAVE_CLD2=1 -I.     -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -pthread -I/usr/include -I/usr/lib64/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DwxDEBUG_LEVEL=0 -DWXUSINGDLL -D__WXGTK__ -pthread  -pthread -I/usr/include/gtkspell-2.0 -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/lucene++ -Wall -Wextra  -DwxNO_UNSAFE_WXSTRING_CONV=1 "-DPOEDIT_PREFIX=\"/usr\"" -c -o extractors/extractor_gettext.o extractors/extractor_gettext.cpp
[   25s] processing ./resources/comment.xrc...
[   25s] processing ./resources/manager.xrc...
[   25s] processing ./resources/menus.xrc...
[   25s] processing ./resources/prefs.xrc...
[   25s] processing ./resources/progress.xrc...
[   25s] processing ./resources/properties.xrc...
[   25s] processing ./resources/summary.xrc...
[   25s] processing ./resources/toolbar.xrc...
[   25s] creating C++ source file /home/abuild/rpmbuild/BUILD/poedit-2.0.2/src/compiled_xrc.cpp...
[   25s] g++ -DPACKAGE_NAME=\"poedit\" -DPACKAGE_TARNAME=\"poedit\" -DPACKAGE_VERSION=\"2.0.2\" -DPACKAGE_STRING=\"poedit\ 2.0.2\" -DPACKAGE_BUGREPORT=\"help@poedit.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"poedit\" -DVERSION=\"2.0.2\" -DHAVE_CXX14=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_THREAD=/\*\*/ -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BOOST_IOSTREAMS=/\*\*/ -DHAVE_MKDTEMP=1 -DHAVE_CLD2_PUBLIC_COMPACT_LANG_DET_H=1 -DHAVE_CLD2=1 -I.     -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -pthread -I/usr/include -I/usr/lib64/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DwxDEBUG_LEVEL=0 -DWXUSINGDLL -D__WXGTK__ -pthread  -pthread -I/usr/include/gtkspell-2.0 -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/lucene++ -Wall -Wextra  -DwxNO_UNSAFE_WXSTRING_CONV=1 "-DPOEDIT_PREFIX=\"/usr\"" -c -o extractors/extractor_legacy.o extractors/extractor_legacy.cpp
[   26s] unicode_helpers.cpp: In function 'TextDirection bidi::get_base_direction(const wxString&)':
[   26s] unicode_helpers.cpp:40:19: error: 'to_icu_raw' is not a member of 'str'
[   26s]      auto s = str::to_icu_raw(text);
[   26s]                    ^~~~~~~~~~
[   26s] make[1]: *** [Makefile:687: unicode_helpers.o] Error 1
[   26s] make[1]: *** Waiting for unfinished jobs....
[   33s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/poedit-2.0.2/src'
[   33s] make: *** [Makefile:466: all-recursive] Error 1
[   33s] error: Bad exit status from /var/tmp/rpm-tmp.i29C1M (%build)
[   33s] 
[   33s] 
[   33s] RPM build errors:
[   33s]     Bad exit status from /var/tmp/rpm-tmp.i29C1M (%build)
[   33s] 
[   33s] argusek failed "build poedit.spec" at Fri Jul  7 18:21:45 UTC 2017
```

unfortunately c++ isn't my language :( I can help with python/perl or little Haskell :)

@rtfontenelle unfortunately not

```
[   63s] g++ -DPACKAGE_NAME=\"poedit\" -DPACKAGE_TARNAME=\"poedit\" -DPACKAGE_VERSION=\"2.0.2\" -DPACKAGE_STRING=\"poedit\ 2.0.2\" -DPACKAGE_BUGREPORT=\"help@poedit.net\" -DPACKAGE_URL=\"\" -DPACKAGE=\"poedit\" -DVERSION=\"2.0.2\" -DHAVE_CXX14=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=/\*\*/ -DHAVE_BOOST_REGEX=/\*\*/ -DHAVE_BOOST_THREAD=/\*\*/ -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BOOST_IOSTREAMS=/\*\*/ -DHAVE_MKDTEMP=1 -DHAVE_CLD2_PUBLIC_COMPACT_LANG_DET_H=1 -DHAVE_CLD2=1 -I.     -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -g -pthread -I/usr/include -I/usr/lib64/wx/include/gtk2-unicode-3.0 -I/usr/include/wx-3.0 -D_FILE_OFFSET_BITS=64 -DwxDEBUG_LEVEL=0 -DWXUSINGDLL -D__WXGTK__ -pthread  -pthread -I/usr/include/gtkspell-2.0 -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/lucene++ -Wall -Wextra  -DwxNO_UNSAFE_WXSTRING_CONV=1 "-DPOEDIT_PREFIX=\"/usr\"" -c -o configuration.o configuration.cpp
[   64s] In file included from /usr/include/wx-3.0/wx/strconv.h:17:0,
[   64s]                  from /usr/include/wx-3.0/wx/strvararg.h:20,
[   64s]                  from /usr/include/wx-3.0/wx/string.h:46,
[   64s]                  from language.h:29,
[   64s]                  from catalog.h:29,
[   64s]                  from cat_sorting.h:29,
[   64s]                  from cat_sorting.cpp:27:
[   64s] /usr/include/wx-3.0/wx/buffer.h: In instantiation of 'static const wxScopedCharTypeBuffer<T> wxScopedCharTypeBuffer<T>::CreateOwned(wxScopedCharTypeBuffer<T>::CharType*, size_t) [with T = char16_t; wxScopedCharTypeBuffer<T>::CharType = char16_t; size_t = long unsigned int]':
[   64s] str_helpers.h:197:37:   required from here
[   64s] /usr/include/wx-3.0/wx/buffer.h:99:27: error: invalid conversion from 'wxScopedCharTypeBuffer<char16_t>::CharType* {aka char16_t*}' to 'const wxChar16* {aka const short unsigned int*}' [-fpermissive]
[   64s]              len = wxStrlen(str);
[   64s]                    ~~~~~~~~^~~~~
[   64s] In file included from /usr/include/wx-3.0/wx/string.h:45:0,
[   64s]                  from language.h:29,
[   64s]                  from catalog.h:29,
[   64s]                  from cat_sorting.h:29,
[   64s]                  from cat_sorting.cpp:27:
[   64s] /usr/include/wx-3.0/wx/wxcrtbase.h:678:32: note:   initializing argument 1 of 'size_t wxStrlen(const wxChar16*)'
[   64s]         WXDLLIMPEXP_BASE size_t wxStrlen(const wxChar16 *s );
[   64s]                                 ^~~~~~~~
[   64s] make[1]: *** [Makefile:687: cat_sorting.o] Error 1
[   64s] make[1]: *** Waiting for unfinished jobs....
[   69s] In file included from /usr/include/wx-3.0/wx/strconv.h:17:0,
[   69s]                  from /usr/include/wx-3.0/wx/strvararg.h:20,
[   69s]                  from /usr/include/wx-3.0/wx/string.h:46,
[   69s]                  from /usr/include/wx-3.0/wx/memory.h:15,
[   69s]                  from /usr/include/wx-3.0/wx/object.h:19,
[   69s]                  from /usr/include/wx-3.0/wx/utils.h:18,
[   69s]                  from catalog.cpp:27:
[   69s] /usr/include/wx-3.0/wx/buffer.h: In instantiation of 'static const wxScopedCharTypeBuffer<T> wxScopedCharTypeBuffer<T>::CreateOwned(wxScopedCharTypeBuffer<T>::CharType*, size_t) [with T = char16_t; wxScopedCharTypeBuffer<T>::CharType = char16_t; size_t = long unsigned int]':
[   69s] str_helpers.h:197:37:   required from here
[   69s] /usr/include/wx-3.0/wx/buffer.h:99:27: error: invalid conversion from 'wxScopedCharTypeBuffer<char16_t>::CharType* {aka char16_t*}' to 'const wxChar16* {aka const short unsigned int*}' [-fpermissive]
[   69s]              len = wxStrlen(str);
[   69s]                    ~~~~~~~~^~~~~
[   69s] In file included from /usr/include/wx-3.0/wx/string.h:45:0,
[   69s]                  from /usr/include/wx-3.0/wx/memory.h:15,
[   69s]                  from /usr/include/wx-3.0/wx/object.h:19,
[   69s]                  from /usr/include/wx-3.0/wx/utils.h:18,
[   69s]                  from catalog.cpp:27:
[   69s] /usr/include/wx-3.0/wx/wxcrtbase.h:678:32: note:   initializing argument 1 of 'size_t wxStrlen(const wxChar16*)'
[   69s]         WXDLLIMPEXP_BASE size_t wxStrlen(const wxChar16 *s );
[   69s]                                 ^~~~~~~~
[   69s] make[1]: *** [Makefile:687: catalog.o] Error 1
[   69s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/poedit-2.0.2/src'
[   69s] make: *** [Makefile:466: all-recursive] Error 1
[   69s] error: Bad exit status from /var/tmp/rpm-tmp.z02I9S (%build)

```

btw.. wxWidgets-3.0.3 so I expect now is problem

> - Don't pass wxString::c_str() to vararg functions any more! Due to Unicode
>   changes, c_str() now returns an object and not a simple char pointer, so
>   using it with vararg functions does not work any more. The preferred
>   solution is to use wx-provided equivalents, e.g. wxPrintf() instead of
>   printf() which has the added benefit that you don't need to call c_str()
>   any more and can pass wxString objects directly. If this is impossible,
>   add explicit casts to "const char*" or "const wchar_t*". See wxString
>   documentation for more details.
> 
>   Notice that using wxString::c_str() with vararg functions does result in
>   compilation errors with g++ and most other compilers but not MSVC. Try to
>   build your code with g++ to detect all such problems at compile-time.

@vslavik reporting problems isn't bossing. Is standard part of QA.

--- a/src/str_helpers.h
+++ b/src/str_helpers.h
@@ -185,7 +185,7 @@ inline std::wstring to_wstring(const icu::UnicodeString& str)
 
     Notice that the resulting string is only valid for the input's lifetime.
  */
-inline wxScopedCharTypeBuffer<UChar> to_icu_raw(const wxString& str)
+inline wxScopedCharTypeBuffer<wxChar16> to_icu_raw(const wxString& str)
 {
     static_assert(U_SIZEOF_UCHAR == 2, "unexpected UChar size");
 #if SIZEOF_WCHAR_T == 2
@@ -194,7 +194,7 @@ inline wxScopedCharTypeBuffer<UChar> to_icu_raw(const wxString& str)
 #else
     auto buf = wxMBConvUTF16().cWC2MB(str.wc_str());
     auto len = buf.length();
-    return wxCharTypeBuffer<UChar>::CreateOwned((UChar*)buf.release(), len);
+    return wxCharTypeBuffer<wxChar16>::CreateOwned((wxChar16*)buf.release(), len);
 #endif
 }
 

--- a/src/unicode_helpers.cpp
+++ b/src/unicode_helpers.cpp
@@ -38,7 +38,7 @@ TextDirection get_base_direction(const wxString& text)
         return TextDirection::LTR;
 
     auto s = str::to_icu_raw(text);
-    switch (ubidi_getBaseDirection(s.data(), (int)s.length()))
+    switch (ubidi_getBaseDirection((const UChar*)s.data(), (int)s.length()))
     {
         case UBIDI_RTL:
             return TextDirection::RTL;

